### DISCUSSION TITLE
Android controls and DI?

### DISCUSSION THREAD

**Discussion by Petrarca181** [AvaloniaUI/Avalonia#11924] [Q&A]
Hello, I'm currently experimenting with Avalonia. I'm wondering if there are any collections of controls specifically designed for mobile platforms, such as Shell, SwipeView, and ContentView. Specifically, I am in need of SwipeView.

Additionally, I am curious about Dependency Injection. I will be using Community MVVM. Would it be a good idea to utilize Microsoft.Extensions.Hosting?

**Comment by maxkatz6** [comment]
> I am curious about Dependency Injection

It's parallel to the UI framework, as Avalonia doesn't force you to use any. You can use Microsoft.Extensions.**DependencyInjection**, AutoFac or any other. You can inject services into view models, for example. Microsoft.Extensions.Hosting is really unnecessary for most applications, especially non-server apps, but you can find example of using it with Avalonia too.
